compiler/ilgen/OMRIlBuilder.cpp

```diff
@@ -1640,6 +1679,52 @@ OMR::IlBuilder::IfOr(TR::IlBuilder **anyTrueBuilder, TR::IlBuilder **allFalseBui
    setComesBack();
    }
 
+/**
+ * @brief Overload taking a varargs instead of an array of JBCondition pointers
+ *
+ * @param anyTrueBuilder builder containing operations to execute if any conditional test 
+ *        evaluates to true (automatically allocated if pointed-to pointer is null)
+ * @param allFalseBuilder builder containing operations to execute if all conditional
+ *        tests are false (automatically allocated if pointed-to pointer is null)
+ * @param numTerms the number of conditional terms
+ * @param ... for each term, provide a TR::IlBuilder::JBCondition object for the condition
+ *
+ * Example:
+ * TR::IlBuilder *cond1Builder = OrphanBuilder();
+ * TR::IlValue *cond1 = cond1Builder->LessThan(
+ *                      cond1Builder->   Load("x"),
+ *                      cond1Builder->   Load("lower"));
+ * TR::IlBuilder *cond2Builder = OrphanBuilder();
+ * TR::IlValue *cond2 = cond2Builder->GreaterOrEqual(
+ *                      cond2Builder->   Load("x"),
+ *                      cond2Builder->   Load("upper"));
+ * TR::IlBuilder *inRange = NULL, *outOfRange = NULL;
+ * IfOr(&outOfRange, &inRange, 2, MakeCondition(cond1Builder, cond1), MakeCondition(cond2Builder, cond2));
+ */
+void
+OMR::IlBuilder::IfOr(TR::IlBuilder **anyTrueBuilder, TR::IlBuilder **allFalseBuilder, int32_t numTerms, ...)
+   {
+   JBCondition **terms = (JBCondition **) _comp->trMemory()->allocateHeapMemory(numTerms * sizeof(JBCondition *));
+   TR_ASSERT(NULL != terms, "out of memory");
+
+   va_list args;
+   va_start(args, numTerms);
+   for (int32_t c = 0; c < numTerms; ++c)
+      {
+      terms[c] = va_arg(args, JBCondition *);
+      }
+   va_end(args);
+
+   IfOr(anyTrueBuilder, allFalseBuilder, numTerms, terms);
+   }
+
+TR::IlBuilder::JBCondition *
+OMR::IlBuilder::MakeCondition(TR::IlBuilder *conditionBuilder, TR::IlValue *conditionValue)
+   {
+   TR_ASSERT(conditionBuilder != NULL, "MakeCondition needs to have non-null conditionBuilder");
```

Comment:
should assert on null conditionValue as well...

Reply:
Yes... yes it should. Fixed.